### PR TITLE
[NTOS] Fix 64 bit warnings

### PR DESCRIPTION
**Pull request by tkreuzer**
Fix MSVC 64 bit warnings

### CODE REVIEW

File: ntoskrnl/ex/interlocked.c [hunk 1]
```diff
@@ -184,8 +184,8 @@ ExInterlockedRemoveHeadList(
         /* Remove the first entry from the list head */
         ListEntry = RemoveHeadList(ListHead);
 #if DBG
-        ListEntry->Flink = (PLIST_ENTRY)0xBADDD0FF;
-        ListEntry->Blink = (PLIST_ENTRY)0xBADDD0FF;
+        ListEntry->Flink = (PLIST_ENTRY)(ULONG_PTR)0xBADDD0FFBADDD0FFULL;
```
Comment (HBelusca) [inline ntoskrnl/ex/interlocked.c]: Can we have these debug patterns in some defines?

File: ntoskrnl/io/pnpmgr/pnpinit.c [hunk 2]
```diff
@@ -63,7 +63,7 @@ PiInitCacheGroupInformation(VOID)
 
         /* Bogus data */
         PiInitGroupOrderTableCount = 0;
-        PiInitGroupOrderTable = (PVOID)0xBABEB00B;
+        PiInitGroupOrderTable = (PVOID)(ULONG_PTR)0xBABEB00BBABEB00BULL;
```
Comment (HBelusca) [inline ntoskrnl/io/pnpmgr/pnpinit.c]: Yeah xD

File: ntoskrnl/mm/ARM3/procsup.c [hunk 3]
```diff
@@ -855,10 +855,10 @@ MiInitializeWorkingSetList(IN PEPROCESS CurrentProcess)
     MmWorkingSetList->HashTable = NULL;
     MmWorkingSetList->HashTableSize = 0;
     MmWorkingSetList->NumberOfImageWaiters = 0;
-    MmWorkingSetList->Wsle = (PVOID)0xDEADBABE;
+    MmWorkingSetList->Wsle = (PVOID)(ULONG_PTR)0xDEADBABEDEADBABEULL;
```
Comment (HBelusca) [inline ntoskrnl/mm/ARM3/procsup.c]: Poor babe ^^

File: ntoskrnl/cache/section/swapout.c [hunk 4]
```diff
@@ -254,7 +254,7 @@ MmFinalizeSectionPageOut(PMM_SECTION_SEGMENT Segment,
     }
 
     /* Note: Writing may evict the segment... Nothing is guaranteed from here down */
-    MiSetPageEvent(Segment, FileOffset->LowPart);
+    MiSetPageEvent(Segment, (ULONG_PTR)FileOffset->QuadPart);
```
Comment (HBelusca) [inline ntoskrnl/cache/section/swapout.c]: So if I understand correctly, when compiled for x86 32-bit, the QuadPart would be truncated?

File: ntoskrnl/io/pnpmgr/pnpinit.c (hunk 2 above)
Comment (ThFabba) [reply ntoskrnl/io/pnpmgr/pnpinit.c]: Yeah, at some point we should change these things to be more neutral. I understand not doing that as part of this though.

File: ntoskrnl/ex/interlocked.c (hunk 1 above)
Comment (ThFabba) [reply ntoskrnl/ex/interlocked.c]: I don't see what that would achieve.

File: ntoskrnl/cache/section/swapout.c (hunk 4 above)
Comment (tkreuzer) [reply ntoskrnl/cache/section/swapout.c]: yes

Comment (HBelusca) [reply ntoskrnl/cache/section/swapout.c]: Is it intentional? (aka. is it expected that on 32-bit, the FileOffset will always hold on 32-bit (which is wrong I think) and/or the rest of the FileOffset will actually be taken into account in some way or another?

**Pull request merged**